diff --git a/ogr/CMakeLists.txt b/ogr/CMakeLists.txt
index ce665225df..b3e07b34bf 100644
--- a/ogr/CMakeLists.txt
+++ b/ogr/CMakeLists.txt
@@ -194,6 +194,7 @@ add_custom_target(check_swq_parser_md5 ALL
                   DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/swq_parser.y"
                   VERBATIM)
 
+if(OFF)
 find_package(BISON)
 add_custom_target(generate_swq_parser
                   COMMAND ${CMAKE_COMMAND}
@@ -202,3 +203,4 @@ add_custom_target(generate_swq_parser
                       -P "${CMAKE_CURRENT_SOURCE_DIR}/generate_swq_parser.cmake"
                   WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                   VERBATIM)
+endif()
diff --git a/ogr/ogrsf_frmts/ods/CMakeLists.txt b/ogr/ogrsf_frmts/ods/CMakeLists.txt
index f2090be63a..1643d1fa3e 100644
--- a/ogr/ogrsf_frmts/ods/CMakeLists.txt
+++ b/ogr/ogrsf_frmts/ods/CMakeLists.txt
@@ -42,6 +42,7 @@ add_custom_target(check_ods_formula_parser_md5 ALL
                   DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/ods_formula_parser.y"
                   VERBATIM)
 
+if(OFF)
 find_package(BISON)
 add_custom_target(generate_ods_formula_parser
                   COMMAND ${CMAKE_COMMAND}
@@ -50,3 +51,4 @@ add_custom_target(generate_ods_formula_parser
                       -P "${CMAKE_CURRENT_SOURCE_DIR}/generate_ods_formula_parser.cmake"
                   WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
                   VERBATIM)
+endif()
